pkg/i2gw/providers/ingressnginx/resource_reader.go

```diff
@@ -34,12 +41,51 @@ func newResourceReader(conf *i2gw.ProviderConf) *resourceReader {
 	}
 }
 
-func (r *resourceReader) ReadResourcesFromCluster(_ context.Context) error {
-	// ingress-nginx does not have any CRDs.
-	return nil
+func (r *resourceReader) readResourcesFromCluster(ctx context.Context) (*storage, error) {
+	storage := newResourcesStorage()
+
+	var ingressList networkingv1.IngressList
+	err := r.conf.Client.List(ctx, &ingressList)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ingresses from the cluster: %w", err)
+	}
+
+	for i, ingress := range ingressList.Items {
+		if common.GetIngressClass(ingress) != NginxIngressClass {
+			continue
+		}
+		storage.Ingresses[types.NamespacedName{Namespace: ingress.Namespace, Name: ingress.Name}] = &ingressList.Items[i]
+	}
+
+	return storage, nil
 }
 
-func (r *resourceReader) ReadResourcesFromFile(_ context.Context, _ string) error {
-	// ingress-nginx does not have any CRDs.
-	return nil
+func (r *resourceReader) readResourcesFromFile(_ context.Context, filename string) (*storage, error) {
```

Comment:
I'd suggest to move out `readResourcesFromFile` logic to a `common` package with the filter option, because this logic is used by both `ingressnginx` & `kong` providers exactly.

Reply:
same comment as above